tests/sum-of-c1-squares.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2009 by Thomas Moulard, AIST, CNRS, INRIA.
+//
+// This file is part of the roboptim.
+//
+// roboptim is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// roboptim is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with roboptim.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "shared-tests/fixture.hh"
+
+#include <iostream>
+
+#include <roboptim/core/io.hh>
+#include <roboptim/core/sum-of-c1-squares.hh>
+
+using namespace roboptim;
+
+struct Null : public DifferentiableFunction
+{
+  Null () : DifferentiableFunction (1, 1, "null function")
```

Comment:
The difference with the `twice-derivable-function` test is that here the values themselves matter (in the other test, you directly provide the dummy values), since we expect an actual result once the function is processed as a squared sum. If you have a null function, you won't detect errors like a missing square in the sum etc.